application/Synchronization/AmazonSynchronizer.cs

```diff
@@ -125,10 +139,14 @@ private void ExecuteAndWait(Action action, int delayInMs = 1000)
             }
             catch (Exception ex)
             {
-                _logger.WriteEntry($"@ExecuteAndWait : {ex.GetLastErrorMessage()}", LoggingLevel.Error, _updatesCount);
+                _telemetryClient.TrackException(ex, new Dictionary<string, string>()
+                {
+                    {"Source", "ExecuteAndWait" },
+                    {"Update#" ,_updatesCount.ToString()}
+                });
             }
 
-            _logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
+            //_logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
```

Comment:
I think it's done using TrackDependency or something like that.

I don't know if the entries have a timestamp.if so you don't need to record the timing.